### PR TITLE
fix(react-grid): clean drag-drop grouping variables

### PR DESCRIPTION
**Pull request by MaximKudriavtsev**

### CODE REVIEW

File: packages/dx-react-grid/docs/reference/grouping-panel.md [hunk 1]
```diff
@@ -43,6 +43,7 @@ Field | Type | Description
 ------|------|------------
 column | [Column](#column) | Specifies the column associated with the item.
 draft | boolean | Specifies whether the item should be rendered for the preview.
+mode | string | Specifies grouping mode "add", "remove" or "reorder".
```
Comment (kvet) [inline packages/dx-react-grid/docs/reference/grouping-panel.md]: It feels a bit strange that mode is linked to the draft state but has such generic name. It also seems that draft and mode should be combined into the one property.

File: packages/dx-react-grid/docs/reference/grouping-panel.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ A value with the following shape:
 Field | Type | Description
 ------|------|------------
 column | [Column](#column) | Specifies the column associated with the item.
-draft | boolean | Specifies whether the item should be rendered for the preview.
+draft | string | Specifies pre-grouping mode "add", "remove" or "reorder".
```
Comment (kvet) [inline packages/dx-react-grid/docs/reference/grouping-panel.md]: May be it should be 'draft? | string'?

File: packages/dx-react-grid-bootstrap3/src/templates/group-panel-item.jsx [hunk 3]
```diff
@@ -85,7 +85,10 @@ GroupPanelItem.propTypes = {
   column: PropTypes.shape({
     title: PropTypes.string,
   }).isRequired,
-  draft: PropTypes.bool,
+  draft: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.bool,
```
Comment (gsobolev) [inline packages/dx-react-grid-bootstrap3/src/templates/group-panel-item.jsx]: It looks like the 'bool' type is redundant.

File: packages/dx-react-grid-material-ui/src/templates/group-panel-item.jsx [hunk 4]
```diff
@@ -71,7 +71,10 @@ GroupPanelItemBase.propTypes = {
   column: PropTypes.shape({
     title: PropTypes.string,
   }).isRequired,
-  draft: PropTypes.bool,
+  draft: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.bool,
```
Comment (gsobolev) [inline packages/dx-react-grid-material-ui/src/templates/group-panel-item.jsx]: Again for the 'bool' type.

File: packages/dx-react-grid/src/components/group-panel-layout.jsx [hunk 5]
```diff
@@ -51,8 +55,14 @@ export class GroupPanelLayout extends React.PureComponent {
       this.setState({ targetColumnIndex });
     };
     this.onLeave = () => {
-      const { draftGroupingChange } = this.props;
+      const { draftGroupingChange, groupingPanelItems } = this.props;
       const { sourceColumnName } = this.state;
+      const sourceItem = groupingPanelItems.filter(item =>
+        item.column.name === sourceColumnName)[0];
+      if (sourceItem && sourceItem.draft === GROUP_ADD_MODE) { // draft -> new_name
```
Comment (gsobolev) [inline packages/dx-react-grid/src/components/group-panel-layout.jsx]: Forgotten comment?

File: packages/dx-react-grid/src/components/group-panel-layout.jsx [hunk 6]
```diff
@@ -177,7 +187,10 @@ GroupPanelLayout.propTypes = {
   changeSortingDirection: PropTypes.func,
   groupingPanelItems: PropTypes.arrayOf(PropTypes.shape({
     column: PropTypes.object,
-    draft: PropTypes.bool,
+    draft: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.bool,
```
Comment (gsobolev) [inline packages/dx-react-grid/src/components/group-panel-layout.jsx]: Again for 'bool'.

File: packages/dx-react-grid/src/components/group-panel-layout.test.jsx [hunk 7]
```diff
@@ -301,5 +305,40 @@ describe('GroupPanelLayout', () => {
       expect(cancelGroupingChange)
         .toHaveBeenCalledTimes(1);
     });
+
+    it('should call cancelGroupingChange on drag leave in the "add" draft', () => {
```
Comment (gsobolev) [inline packages/dx-react-grid/src/components/group-panel-layout.test.jsx]: '...when the draft is "add"'

File: packages/dx-grid-core/src/plugins/table-group-row/computeds.js [hunk 8]
```diff
@@ -1,5 +1,9 @@
 import { TABLE_DATA_TYPE } from '../table-view/constants';
 import { TABLE_GROUP_TYPE } from './constants';
+import { GROUP_ADD_MODE, GROUP_REMOVE_MODE } from '../grouping-state/constants';
+
+const isDraft = columnDraftGrouping => columnDraftGrouping.draft === GROUP_REMOVE_MODE
```
Comment (kvet) [inline packages/dx-grid-core/src/plugins/table-group-row/computeds.js]: Why `isDraft`?

Comment (MaximKudriavtsev) [reply packages/dx-grid-core/src/plugins/table-group-row/computeds.js]: @kvet  `isRemove` ?

Comment (MaximKudriavtsev) [reply packages/dx-grid-core/src/plugins/table-group-row/computeds.js]: @kvet `isGroupingChange` ?

**Pull request merged**